File: server/app/schema/role.py
from pydantic import BaseModel
from typing import List
from .permission import PermissionResponse

class RoleBase(BaseModel):
    name:str

class RoleResponse(RoleBase):
    id: int
    permissions:List[PermissionResponse]

    class Config:
        from_attributes = True

        
class CreateRole(RoleBase):
    permission_ids:List[int]

    class Config:
        from_attributes :True

File: client/app/schema/role.ts
import { Permission } from "./permission";

export interface Role {
    id: number;
    name: string;
    permissions: Permission[];  
}

export interface CreateRole {
    name: string;
    permission_ids: number[]; 
}

export interface UpdateRole {
    name?: string; 
    permission?: Permission[]; 
}